java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -265,33 +408,78 @@ public HostColumnVector copyToHost() {
       HostMemoryBuffer hostDataBuffer = null;
       HostMemoryBuffer hostValidityBuffer = null;
       HostMemoryBuffer hostOffsetsBuffer = null;
-      BaseDeviceMemoryBuffer valid = offHeap.getValid();
-      BaseDeviceMemoryBuffer offsets = offHeap.getOffsets();
-      BaseDeviceMemoryBuffer data = offHeap.getData();
+      BaseDeviceMemoryBuffer valid = (BaseDeviceMemoryBuffer)getValidityBuffer();
+      BaseDeviceMemoryBuffer offsets = (BaseDeviceMemoryBuffer)getOffsetBuffer();
+      BaseDeviceMemoryBuffer data;
+      DType type = this.type;
+      Long rows = this.rows;
+      // hardcoded for lists for now
+      ColumnViewPointerAccess leafChildWithData = getChildColumnView(0);
+      while (leafChildWithData != null && leafChildWithData.getChildColumnView(0) != null) {
```

Comment:
Can we have an explanation as to why we need to go two levels deep if there are two levels, but only one if there is just one?

Reply:
Added some info let me know if that makes sense.

Reply:
I still don't understand why data is different. I understand that we are treating offsets differently and so we have to play around with the number of children and offsets into the children, but data should just be in the leaf column right? If we play the same game that we did for Strings don't we risk losing the validity for that leaf column when we do this? If that is what we are doing here then I would argue that eventually we should just make string match list.